src/main/kotlin/com/hj/leetcode/kotlin/problem1802/Solution.kt

```diff
@@ -0,0 +1,72 @@
+package com.hj.leetcode.kotlin.problem1802
+
+/**
+ * LeetCode page: [1802. Maximum Value at a Given Index in a Bounded Array](https://leetcode.com/problems/maximum-value-at-a-given-index-in-a-bounded-array/);
+ */
+class Solution {
+    /* Complexity:
+     * Time O(Log(maxSum)) and Space O(1);
+     */
+    fun maxValue(n: Int, index: Int, maxSum: Int): Int {
+        require(n >= 1)
+        require(index in 0 until n)
+        require(maxSum >= n)
+
+        /* The idea is to form the resulting array using a greedy approach, where the value at
+         * each index i is calculated by subtracting the distance between index i and the given
+         * index from the value at the given index, with the resulting value constrained to be
+         * at least one.
+         *
+         * Since the sum of the resulting array is directly proportional to the value at the
```

Comment:
"directly proportional" seems not to be a correct description of their relationship

Reply:
Resolved at https://github.com/hj-core/hj-leetcode-kotlin/pull/84/commits/aa3beef3283c37436836ee6af79a6b456f10244b